src/lib/client.js

```diff
@@ -3,23 +3,27 @@
 const assert = require('assert')
 const WebSocket = require('ws')
 const debug = require('debug')('ilp-plugin-payment-channel-framework:client')
-const url = require('url')
+const { URL } = require('url')
 
 module.exports = class BtpClient {
-  constructor ({ server, secret, plugin, insecure }) {
-    this._server = server
+  constructor ({ btpUri, plugin }) {
```

Comment:
why are we no longer allowing the token to be passed in independently of the URI?

Reply:
Because we have a BTP URI format now, right? If you specify the token separately, then either it would match the token from the btpUri, or be incorrect, and in both cases passing the token independently would be useless.

If you pass the token separately and then also remove it from your btpUri, that would mean your btpUri would by itself be incorrect, so that would also not be a very useful move?

Reply:
I agree with you if we're saying the BTP URI is the new version of that `ilp_secret` connection string, where it's almost treated as an opaque token. But from our discussion, I thought that the BTP URI was more like a database connection string; a convenient shortcut to combine different details, but not canonical.

Reply:
I think the plugin-framework constructor options will be *the* place where BTP URIs are used. If we don't use them there, then where will we?